File: src/main/scala/nu/tengstrand/tetrisanalyzer/piecegenerator/JavaPieceGenerator.scala
package nu.tengstrand.tetrisanalyzer.piecegenerator

import nu.tengstrand.tetrisanalyzer.piece.Piece
import java.util.Random

/**
 * Uses the standard random generator in Java.
 */
class JavaPieceGenerator  extends PieceGenerator {
  private val random = new Random
  private var number = nextRandomPieceNumber

  private def nextRandomPieceNumber = random.nextInt(Piece.NumberOfPieceTypes) + 1

  def pieceNumber = number

  def prepareNext() { number = nextRandomPieceNumber }
}

File: otherlang/src/main/java/com/github/tetrisanalyzer/piecegenerator/JavaPieceGenerator.java
package nu.tengstrand.tetrisanalyzer.piecegenerator;

import nu.tengstrand.tetrisanalyzer.piece.Piece;

import java.util.Random;

/**
 * Uses the standard random generator in Java.
 */
public class JavaPieceGenerator extends PieceGenerator {
    private Random random = new Random();

    @Override
    public int nextPieceNumber() {
        return random.nextInt(Piece.NUMBER_OF_PIECE_TYPES) + 1;
    }
}